staff/nicolas-martin/horroy/scripts/horroy.js

```diff
@@ -0,0 +1,209 @@
+    for (var i = 0; i < arguments.length; i++)
+        this[i] = arguments[i];
+}
+
+/**
+* Tests is anything is a horroy Type
+* 
+* @param {*} horr - anything to test if is a horroy
+* @returns {boolean} - true if horr is horroy type and false otherwise
+*/
+
+Horroy.isHorroy = function (horr) {
+    return (horr instanceof Horroy);
+}
+
+/**
+* Creates a new horroy where every item will be fill with every argument
+* 
+* @param {*} - arguments to creates the horroy
+* @returns {horroy} - horroy returned with all items in arguments
+*/
+
+Horroy.of = function () {
+    var horr = new Horroy;
+    for (var i = 0; i < arguments.length; i++) {
+        horr.push(arguments[i]); 
+    }
+    return horr;
+}
+
+/**
+* Creates a new Horroy copy from an array-like or iterable object.
+* 
+* @param {*} - value to split and creates the horroy
+* @returns {horroy} - a new horroy instance
+*/
+
+Horroy.from = function (value) {
+    var horr = new Horroy;
+    horr.length = value.length;
+    if (typeof value === 'string')
+        for (var i = 0; i < value.length; i++)
+            horr[i] = value[i];
+    return horr;
+};
+
+/**
+ * JOIN - Creates and returns a new string by concatenating all of the elements in a Horroy
+ * with separator between them
+ * 
+ * @param {*} sep - will be converted to string. If is undefined a comma sepator will be used
+ * @returns {string} result - string resulting of concatenate all items. If horroy is empty 
+ * an empty string will be returned
+ *
+ */
+
+Horroy.prototype.join = function(sep) {
+    var separator = (sep !== undefined) ? String(sep) : (!this.length) ? '' : ',';
+    var result = '';
+    for (var i = 0; i < this.length; i++) 
+        result += (i !== this.length-1) ? this[i] + separator : this[i];  
+    return result;
+}
+
+/**
+ * Returns the value of the first item in the horroy that satisfies the callback function. 
+ * Otherwise undefined is returned 
+ * 
+ * @param {function} callback - function with condition to test in every item
+ * @returns {*} first item found that satisfy the callback function
+ * @throws {TypeError} if callback is not a function
+ */
+
+Horroy.prototype.find = function (callback) {
+    if (typeof callback !== 'function') throw TypeError(callback + ' is not a function');
+    for (var i = 0; i < this.length; i++){
+        if (callback(this[i])) return this[i]
+    }
+    return undefined;
+}
+
+/**
+ * Creates a new array with all elements that pass the test implemented by callback. 
+ * 
+ * @param {function} callback - function with the test
+ * @returns {horroy} horroy with items that passed the condition implemented by callback function
+ * @throws {TypeError} if callback is not a function
+ */
+
+Horroy.prototype.filter = function(callback) {
+    if (typeof callback !== 'function') throw TypeError(callback + ' is not a function');
+    var horr = new Horroy;
+    for (var i = 0; i < this.length; i++)
+        if (callback(this[i])) horr.push(this[i])
```

Comment:
It would be better if you try not to use a method inside another method, so to do that filter without using push

Reply:
fixed!